lib/rdoc/code_object/context/section.rb

```diff
@@ -214,19 +172,8 @@ def plain_html
   # +comment+
 
   def remove_comment comment
-    return if @comments.empty?
-
-    case @comments
-    when Array then
-      @comments.delete_if do |my_comment|
-        my_comment.file == comment.file
-      end
-    when RDoc::Markup::Document then
-      @comments.parts.delete_if do |document|
-        document.file == comment.file.name
-      end
-    else
-      raise RDoc::Error, "BUG: unknown comment class #{@comments.class}"
+    @comments.delete_if do |my_comment|
```

Comment:
Can we also rename these locals to something like `stored_comment` and `target_comment`?

```rb
@comments.delete_if do |stored_comment|
  stored_comment.file == target_comment.file
end
```